### ISSUE TITLE
DisplayServer.window_is_focused returns true for unfocused window after calling DisplayServer.window_set_flag

### ISSUE THREAD

**Issue by chelovekbeznika** [godotengine/godot#89268]
### Tested versions

- Reproducible in v4.2.1.stable.official [b09f793f5]

### System information

Godot v4.2.1.stable - Windows 10.0.22631 - GLES3 (Compatibility) - NVIDIA GeForce RTX 3090 (NVIDIA; 31.0.15.5161) - 12th Gen Intel(R) Core(TM) i9-12900K (24 Threads)

### Issue description

DisplayServer.window_is_focused returns true for unfocused window after calling DisplayServer.window_set_flag. I expect DisplayServer.window_is_focused to return false when window is out of focus.

### Steps to reproduce

1. Start MRP
2. Press "ui_accept" event (space on keyboard)
3. Quickly switch to other window (on second monitor if possible, it's easier to observe)
4. Wait 5+ seconds
5. Stop MRP
6. Observer log in editor with dedup

### Minimal reproduction project (MRP)

[WindowFlagsBugMRP.zip](https://github.com/godotengine/godot/files/14531620/WindowFlagsBugMRP.zip)

**Comment by Sauermann** [comment]
Have tested the MRP on Linux X11 Xfce with v4.3.dev.custom_build [aef11a142] and I am unable to reproduce the issue.

**Comment by luevano** [comment]
Tested the MRP in Windows 11, can confirm in `v4.2.1.stable.official` [b09f793f5], `v4.3.dev4.official` [df78c0636] and will compile til latest commit (will update this comment).

**Comment by chelovekbeznika** [comment]
Same thing goes for notifications (NOTIFICATION_WM_WINDOW_FOCUS_IN and NOTIFICATION_WM_WINDOW_FOCUS_OUT). Updated MRP to illustrate.
[WindowFlagsBugMRP.zip](https://github.com/godotengine/godot/files/14544162/WindowFlagsBugMRP.zip)

**Comment by luevano** [comment]
> Same thing goes for notifications (NOTIFICATION_WM_WINDOW_FOCUS_IN and NOTIFICATION_WM_WINDOW_FOCUS_OUT). Updated MRP to illustrate. [WindowFlagsBugMRP.zip](https://github.com/godotengine/godot/files/14544162/WindowFlagsBugMRP.zip)

Yeah, those are behaving correctly; those notifications are sent because the focus was set or killed based on `window_set_flag` internal (wrong?) behavior.

I've been looking into this and I was able to apply the "desired" behavior but I'm not sure if that's correct. I'm not familiar with the windows api and I'm not sure why it was set to behave like this in the first place (meaning it could be a bug or just intended due to windows api).

I can open a PR for a "fix", but i'm not sure if this needs "fixing" or is just a thing with windows; I also don't know what side effects this could cause yet.

**Comment by luevano** [comment]
For reference, to "fix" this I changed:

https://github.com/godotengine/godot/blob/0ace0a129284ffc6646b199699c1607a316fcec0/platform/windows/display_server_windows.cpp#L1957

To `SetWindowPos(wd.hWnd, wd.always_on_top ? HWND_TOPMOST : HWND_NOTOPMOST, 0, 0, 0, 0, SWP_FRAMECHANGED | SWP_NOMOVE | SWP_NOSIZE | SWP_NOACTIVATE);`

But this is just a hack, as the root cause feels like it is something else. I think this should be set to `SWP_NOACTIVATE` and then control the window activation somewhere else. For example, in (which is called 2 lines after calling `_update_window_style`):

https://github.com/godotengine/godot/blob/0ace0a129284ffc6646b199699c1607a316fcec0/platform/windows/display_server_windows.cpp#L2101

This should actually control whether the window should activate (focus) but setting it to `SW_SHOWNA` doesn't seem to work.

cc @bruvzg as you were the last one working on this code (not saying you introduced the "bug") and it looks like you're familiar with these apis and their behavior/intended behavior?
